R/schema_SheetProperties.R

```diff
@@ -0,0 +1,43 @@
+# https://developers.google.com/sheets/api/reference/rest/v4/spreadsheets/sheets#SheetProperties
+new_SheetProperties <- function(sheetId = NULL,
+                                title = NULL,
+                                index = NULL,
+                                sheetType = NULL,
+                                gridProperties = NULL,
+                                hidden = NULL,
+                                tabColor = NULL,
+                                rightToLeft = NULL) {
+  x <- list(
+    sheetId = sheetId,
+    title = title,
+    index = index,
+    sheetType = sheetType,
+    gridProperties = gridProperties,
+    hidden = hidden,
+    tabColor = tabColor,
+    rightToLeft = rightToLeft
+  )
+  structure(validate_SheetProperties(x), class = "SheetProperties")
+}
+
+validate_SheetProperties <- function(x) {
+  maybe_non_negative_integer(x$sheetId, "sheetId")
+  maybe_string(x$title, "title")
+  maybe_non_negative_integer(x$index, "index")
+  maybe_string(x$sheetType, "sheetType") # enum
```

Comment:
In addition to ~200 schemas there are *lots* of enums. They aren't particularly easy to pull out of the discovery document in a coherent way. But you could imagine doing so, in a fit of recursive fury, and create a way to declare something a enum, check supplied values against the valid ones, etc.

Reply:
Auto-generating an internal data structure and code around the enums sounds much easier to me (maybe even a clear "yes, do this"?) than doing same for the schemas.